puppet/lib/puppet/util/puppetdb/command.rb

```diff
@@ -42,7 +42,7 @@ def initialize(command, version, certname, payload)
       # Puppet 4.1.0. We need a better answer to non-utf8 data end-to-end.
       }.to_pson, "Error encoding a '#{command}' command for host '#{certname}'")
     end
-    @command = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(command).gsub(" ", "_")
+    @command = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(command)
```

Comment:
I don't understand this change... what did the `gsub` used to do? It looks to me like it already tried to submit commands with "_" instead of space.